pkg/memory/memory_linux.go

```diff
@@ -64,19 +64,30 @@ func AreaForNode(ctx *context.Context, nodeID int) (*Area, error) {
 		fmt.Sprintf("node%d", nodeID),
 	)
 
-	blockSizeBytes, err := memoryBlockSizeBytes(paths.SysDevicesSystemMemory)
-	if err != nil {
-		return nil, err
-	}
+	var err error
+	var blockSizeBytes uint64
+	var totPhys int64
+	var totUsable int64
 
-	totPhys, err := memoryTotalPhysicalBytesFromPath(path, blockSizeBytes)
+	totUsable, err = memoryTotalUsableBytesFromPath(filepath.Join(path, "meminfo"))
 	if err != nil {
 		return nil, err
 	}
 
-	totUsable, err := memoryTotalUsableBytesFromPath(filepath.Join(path, "meminfo"))
-	if err != nil {
-		return nil, err
+	blockSizeBytes, err = memoryBlockSizeBytes(paths.SysDevicesSystemMemory)
```

Comment:
Hi @jaypipes , I think should use memTotalPhysicalBytes function here, memTotalPhysicalBytes will scan syslog to get physical size, and check return value > 0 to decide whether to use totUsable memory.  
https://github.com/jaypipes/ghw/blob/227643e14380812d61b941f7aa52d1156b6d900a/pkg/memory/memory_linux.go#L105-L113

Reply:
@wanyaoqi good point. done!